pkg/reconciler/v1alpha1/pipelinerun/pipelinerun.go

```diff
@@ -144,21 +150,74 @@ func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
 
 func (c *Reconciler) reconcile(ctx context.Context, pr *v1alpha1.PipelineRun) error {
 	// fetch the equivelant pipeline for this pipelinerun Run
-	name := pr.Spec.PipelineRef.Name
-	if _, err := c.pipelineLister.Pipelines(pr.Namespace).Get(name); err != nil {
-		c.Logger.Errorf("%q failed to Get Pipeline: %q",
-			fmt.Sprintf("%s/%s", pr.Namespace, pr.Name),
-			fmt.Sprintf("%s/%s", pr.Namespace, name))
-		return nil
+	if _, err := c.getPipelineIfExists(pr); err != nil {
+		return err
 	}
-
 	// TODO fetch the taskruns status for this pipeline run.
 
 	// TODO check status of tasks and update status of PipelineRuns
 
 	return nil
 }
 
+func (c *Reconciler) createPipelineRun(ctx context.Context, p *v1alpha1.Pipeline) error {
+	tasks, err := c.getTasks(p)
+	if err != nil {
+		return errors.NewBadRequest(fmt.Sprintf("pipeline %q is not valid due to %v", p.Name, err))
+	}
+	for _, t := range tasks {
```

Comment:
check my comment in the review, should we create all taskRun or just the first one and taskRun controller can do the rest?